pkg/policy/l4.go

```diff
@@ -39,24 +40,36 @@ func (l7 L7DataMap) MarshalJSON() ([]byte, error) {
 		return []byte("[]"), nil
 	}
 
+	/* First, create a sorted slice of the selectors so we can get
+	 * consistent JSON output */
+	selectors := make(api.EndpointSelectorSlice, len(l7))
+	for es := range l7 {
+		selectors = append(selectors, es)
+	}
+	sort.Sort(selectors)
+
+	/* Now we can iterate the slice and generate JSON entries. */
 	var err error
 	buffer := bytes.NewBufferString("[")
-	for k, v := range l7 {
-		buffer.WriteString("\n  {    \n    \"")
-		buffer.WriteString(k.LabelSelectorString())
-		buffer.WriteString("\": ")
-		b, err := json.MarshalIndent(v, "    ", "  ")
+	for _, es := range selectors {
+		if _, ok := l7[es]; !ok {
```

Comment:
Alright, this is why the tests are passing, you're ignoring the `len(l7)` `nil` selectors that are always at the beginning of your slice. You should be able to remove this after you fix the slice creation above?

Reply:
Nice catch, yup! I'll fix this up.